tests/kyber_nistkats.rs

```diff
@@ -0,0 +1,157 @@
+use libcrux::kem;
+use serde::Deserialize;
+use serde_json;
+
+use libcrux::digest;
+
+use std::{fs::File, io::BufReader};
+
+#[derive(Deserialize)]
+struct KyberNISTKAT {
+    #[serde(with = "hex::serde")]
+    key_generation_seed: [u8; 64],
+
+    #[serde(with = "hex::serde")]
+    sha3_256_hash_of_public_key: [u8; 32],
+
+    #[serde(with = "hex::serde")]
+    sha3_256_hash_of_secret_key: [u8; 32],
+
+    #[serde(with = "hex::serde")]
+    encapsulation_seed: [u8; 32],
+
+    #[serde(with = "hex::serde")]
+    sha3_256_hash_of_ciphertext: [u8; 32],
+
+    #[serde(with = "hex::serde")]
+    shared_secret: [u8; 32],
+}
+
+#[test]
+fn kyber768_nist_known_answer_tests() {
```

Comment:
Let's put these functions in a macro or into one to avoid duplicating the code three times.